src/Model/Contact.php

```diff
@@ -1009,4 +1011,283 @@ public static function updateAvatar($avatar, $uid, $cid, $force = false)
 
 		return $data;
 	}
+
+	/**
+	 * @param integer $id contact id
+	 * @return boolean
+	 */
+	public static function update($id)
```

Comment:
Please rename this function to `Contact::updateFromProbe` as it is unclear from the name and parameter alone.

Reply:
OK, i just saw it as "i'm updating a contact"

Reply:
Usual `update` operation take an array of modified fields to update the according model/row. This is more like a self-update based on an URL probe.